src/data/events.ts

```diff
@@ -9,26 +9,66 @@ import EventImage2 from '@assets/events/projects-event-image-2.jpeg';
 import EventImage3 from '@assets/events/projects-event-image-3.jpeg';
 import type { GetImageResult } from 'astro';
 import * as qs from 'qs';
-import type { Event as EventCMS } from './payload-types.ts';
+import type { Address, Event as EventCMS, Media, Team } from './payload-types.ts';
 
-export interface Event {
+export interface EventCalendarEntry {
   title: string;
   date: Date;
   description: string;
-  content_html?: string;
   link?: string;
-  slug?: string;
-  image: {
-    src: Awaited<ReturnType<typeof getImage>>;
-  };
+  image: YVImage;
 }
 
-export async function getNext3Events(): Promise<[Event, Event, Event]> {
-  const image1 = await getImage({ src: thirdEventImage });
-  const image2 = await getImage({ src: calendarCoverImage });
-  const image3 = await getImage({ src: pastEvent });
+export interface EventPage {
+  title: string;
+  start: Date;
+  end: Date;
+  slug: string;
+  content_html: string;
+  heroImage?: YVImage;
+  address: Address;
+  audience: string;
+  cost: string;
+  team: Team;
+  registrationLink: string;
+  timetable: YVImage;
+}
+
+export type YVImage = {
+  src: GetImageResult;
+  alt: string;
+};
 
+export async function getAllEvents(): Promise<EventPage[]> {
+  const response = await fetch(`${process.env.CMS_URL}/api/events`);
+  const data = await response.json();
+  const events = data.docs as EventCMS[];
+  const eventsWithSlug = events.filter((event) => event.slug);
+  const promises = eventsWithSlug.map(async (event) => ({
+    title: event.title,
+    start: new Date(event.start),
+    end: new Date(event.end),
+    content_html: event.content_html || undefined,
+    slug: `${event.slug}`,
+    heroImage: await getEventImage(event.heroImage?.value),
+    address: event.address,
+    audience: event.audience,
+    cost: event.cost,
+    team: event.team,
+    registrationLink: event.registrationLink || undefined,
+    timetable: await getEventImage(event.timetable?.value),
```

Comment:
You can use the [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax).



```suggestion
    ...event,
    start: new Date(event.start),
    end: new Date(event.end),
    content_html: event.content_html || undefined,
    heroImage: await getEventImage(event.heroImage?.value),
    registrationLink: event.registrationLink || undefined,
    timetable: await getEventImage(event.timetable?.value),
```

Also why use the `event.registrationLink || undefined` for example? Could you please add a small comment explaining why we need this?

Reply:
This syntax is new to me, but looks handy. I'm not sure myself why there was `event.registrationLink || undefined`.